cmd/dockerd/daemon_windows.go

```diff
@@ -100,9 +100,18 @@ func newCgroupParent(config *config.Config) string {
 	return ""
 }
 
-func (cli *daemonCLI) initContainerd(_ context.Context) (func(time.Duration) error, error) {
-	system.InitContainerdRuntime(cli.ContainerdAddr)
-	return nil, nil
+func (cli *daemonCLI) initContainerd(ctx context.Context) (func(time.Duration) error, error) {
+	defer func() { system.EnableContainerdRuntime(cli.ContainerdAddr) }()
+
+	if cli.ContainerdAddr != "" {
+		return nil, nil
+	}
+
+	if cli.DefaultRuntime != config.WindowsV2RuntimeName {
+		return nil, nil
+	}
```

Comment:
I'd suggest inverting this test to be an early-out in non-containerd mode. Long-term, I hope we'll be able to remove all tests against `config.WindowsV2RuntimeName`, and allow any runtime name except `config.WindowsV2RuntimeName` to flow through to containerd.

It's not particularly urgent to do in this PR, since work to enable alternative container runtimes will have to touch a bunch of other places too.

```suggestion
	if cli.DefaultRuntime == config.WindowsV1RuntimeName {
		return nil, nil
	}
```

Reply:
Such condition makes much more sense to me.

Reply:
+1 on this (not a blocker for this PR though)

Reply:
Got it! I'll include this in the refactor of Windows runtime name handling to support additional runtimes.